### PR TITLE
Update CI conventions

### PR DESCRIPTION
**Pull request by NSeydoux**
Apply CI conventions from our other repositories:
- NPM scripts naming scheme
- Audit and Lint workflow separate from the unit tests
- End-to-end tests separate from unit tests

This also fixes audit issues.

### CODE REVIEW

**Review by ajacksified — APPROVED** [review]

**Review by ThisIsMissEm — CHANGES_REQUESTED** [review]
The only one I'd like to request changes on is the docs scripts.

File: package.json [hunk 1]
```diff
@@ -5,14 +5,17 @@
   "license": "MIT",
   "scripts": {
     "build": "rollup --config rollup.config.js",
-    "build-api-docs": "typedoc",
-    "build-docs-preview-site": "npm run build-api-docs; cd docs/api; rm source/api/index.md; make html; cd ../; rm -r dist || true; mkdir -p dist/api; cp -r api/build/html/. dist/;",
-    "lint": "eslint src",
+    "docs:build": "typedoc",
+    "docs:preview": "npm run docs:build; cd docs/api; rm source/api/index.md; make html; cd ../; rm -r dist || true; mkdir -p dist/api; cp -r api/build/html/. dist/;",
```
[the remark this answers is not in the document]
Comment (NSeydoux) [reply package.json:9]: There's no rush for this, I'll revert it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -27,148 +27,7 @@ jobs:
...
-        E2E_TEST_ESS_IDP_URL: ${{ secrets.E2E_TEST_ESS_PROD_IDP_URL }}
-        E2E_TEST_ESS_POD: ${{ secrets.E2E_TEST_ESS_PROD_POD }}
-        E2E_TEST_ESS_COGNITO_USER: ${{ secrets.E2E_TEST_ESS_PROD_COGNITO_USER }}
-        E2E_TEST_ESS_COGNITO_PASSWORD: ${{ secrets.E2E_TEST_ESS_PROD_COGNITO_PASSWORD }}
-        E2E_TEST_DEV_NEXT_POD: ${{ secrets.E2E_TEST_DEV_NEXT_POD }}
-        E2E_TEST_DEV_NEXT_IDP_URL: ${{ secrets.E2E_TEST_DEV_NEXT_IDP_URL }}
-        E2E_TEST_DEV_NEXT_COGNITO_USER: ${{ secrets.E2E_TEST_DEV_NEXT_COGNITO_USER }}
-        E2E_TEST_DEV_NEXT_COGNITO_PASSWORD: ${{ secrets.E2E_TEST_DEV_NEXT_COGNITO_PASSWORD }}
-        GITHUB_OS: ${{ runner.os }}
     - run: npx package-check
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:31]: this is in lint / audit shared action already, iirc.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -27,148 +27,7 @@ jobs:
...
     - name: Archive code coverage results
```
Comment (ThisIsMissEm) [inline .github/workflows/ci.yml:32]: We don't need these two artifacts, they're never reused anywhere.

**Review by ThisIsMissEm — APPROVED** [review]
Two minor comments, but looking good

File: .github/workflows/audit.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+name: Audit
+
+on:
+  push:
+  schedule:
+    - cron: "40 10 * * *"
+
+jobs:
+    audit:
+      uses: inrupt/javascript-style-configs/.github/workflows/reusable-audit.yml@create-reusable-workflows
```
Comment (ThisIsMissEm) [inline .github/workflows/audit.yml:10]: Oh, also, these will break once that PR is merged, side-effect of using something in testing. Let's make sure these work then merge that branch

**Pull request merged**